### PR TITLE
Feat/sentry error boundary

### PR DESCRIPTION
**Pull request by olavoparno**
closes  https://github.com/Stand-With-Crypto/swc-internal/issues/302

fixes 

## What changed? Why?

- add more granularity in terms of errors on our application with error boundaries
- we've added it to the following parts of the app:

1. Donate button
2. Sign in button dialog
3. Actions
4. Map

## How has it been tested?

- [X] Locally
- [x] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

### CODE REVIEW

**Review by eduardopicolo-cb — CHANGES_REQUESTED** [review]
We also need to add the boundary to the actions deeplinks

**Comment by eduardopicolo-cb** [comment]
There is also a new zod plugin for sentry https://github.com/getsentry/sentry-javascript/pull/11144, this should allow for better grouping for validation related errors

And this integration that should add more error data from third-party errors https://docs.sentry.io/platforms/javascript/guides/nextjs/configuration/integrations/extraerrordata/

**Comment by olavoparno** [comment]
> There is also a new zod plugin for sentry [getsentry/sentry-javascript#11144](https://github.com/getsentry/sentry-javascript/pull/11144), this should allow for better grouping for validation related errors
> 
> And this integration that should add more error data from third-party errors https://docs.sentry.io/platforms/javascript/guides/nextjs/configuration/integrations/extraerrordata/

we already have `Sentry.extraErrorDataIntegration()` in place

**Review by lucasrmp-cb — CHANGES_REQUESTED** [review]

File: src/utils/web/errorBoundary/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (extras) {
+          scope.setExtras(extras)
+        }
+
+        scope.setLevel(severityLevel ?? 'warning')
+
+        return scope
+      }}
+      fallback={fallback ?? ErrorFallbackComponent}
```
[the remark this answers is not in the document]
Comment (olavoparno) [reply src/utils/web/errorBoundary/index.tsx]: cool, you're right. removed it

File: src/utils/web/errorBoundary/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (tags) {
+          scope.setTags(tags)
+        }
+
+        if (extras) {
+          scope.setExtras(extras)
+        }
+
+        scope.setLevel(severityLevel ?? 'warning')
```
[the remark this answers is not in the document]
Comment (olavoparno) [reply src/utils/web/errorBoundary/index.tsx]: thanks. adjusted it

**Review by lucasrmp-cb — APPROVED** [review]

**Review by eduardopicolo-cb — APPROVED** [review]

**Pull request merged**